src/data/pokemon/form_change_tables.h

```diff
@@ -1,199 +1,633 @@
-/*
-FORM_ITEM_HOLD:
-    Form change activates when the specified item is given to or taken from the selected Pokémon.
-    Alternatively, form change activates when the specified item is is given to or taken from
-    the selected Pokémon that has a particular ability.
-    param1 = item to hold
-    param2 = ability to check for, optional
-
-FORM_ITEM_USE:
-    Form change activates when the item is used on the selected Pokémon.
-    param1 = item to use
-    param2 = DAY if form change activates in the daytime, optional
-             NIGHT if form change activates at nighttime, optional
-
-FORM_MOVE:
-    Form change activates when the Pokémon learns or forgets the move.
-    param1 = move to check for
-    param2 = WHEN_LEARNED if form change activates when move is forgotten
-             WHEN_FORGOTTEN if form change activates when move is learned
-
-FORM_WITHDRAW:
-    Form change activates when the Pokémon is withdrawn from the PC or Daycare.
-    no parameters
-
-FORM_BATTLE_BEGIN:
-    Form change activates when the Pokémon is sent out at the beginning of a battle
-    param1 = item to hold, optional
-    param2 = a move that will be replaced, optional
-    param3 = a new move to replace it with, optional
-
-FORM_BATTLE_END:
-    Form change activates at the end of a battle
-    param1 = item to hold, optional
-    param2 = a move that will be replaced, optional
-    param3 = a new move to replace it with, optional
-*/
-
-// FORM_MOVE param2 Arguments
-#define WHEN_LEARNED    0
-#define WHEN_FORGOTTEN  1
-
-// FORM_ITEM_USE param2 Arguments
-#define DAY    1
-#define NIGHT  2
+static const struct FormChange sVenusaurFormChangeTable[] = {
+    {FORM_CHANGE_BATTLE_MEGA_EVOLUTION_ITEM,    SPECIES_VENUSAUR_MEGA, ITEM_VENUSAURITE},
+    {FORM_CHANGE_FAINT,                         SPECIES_VENUSAUR},
+    {FORM_CHANGE_END_BATTLE,                    SPECIES_VENUSAUR},
+    {FORM_CHANGE_TERMINATOR},
+};
```

Comment:
I'm surprised you didn't decide to make a macro that automates simple mega evolution! ... Not saying you _should_ do that (it's fine as-is), just that knowing your contributions, I would have expected it :D

Reply:
Other form changes like Gigantamax needed to be considered, so I didn't do that.